addons/sourcemod/scripting/shavit-tas.sp

```diff
@@ -0,0 +1,1702 @@
+															*/
+				}
+
+				if(gB_ProcessFrame[client])
+				{
+					gF_TASTime[client] += GetTickInterval();
+
+					int iFrameNumber = gA_Frames[client].Length + 1;
+					if(gI_IndexCounter[client] + gI_PreFrameCount[client] != iFrameNumber-2)
+					{
+						//UnPaused in different tick
+						iFrameNumber = gI_IndexCounter[client] + 1 + gI_PreFrameCount[client];
+					}
+					gA_Frames[client].Resize(iFrameNumber);
+
+					framedata_t frame;
+
+					frame.fTime = gF_TASTime[client];
+
+					float fAngles[3];
+					GetClientEyeAngles(client, fAngles);
+					frame.fEyeAngles[0] = fAngles[0];
+					frame.fEyeAngles[1] = fAngles[1];
+
+					GetEntPropVector(client, Prop_Send, "m_vecOrigin", frame.fPosition);
+					GetEntPropVector(client, Prop_Data, "m_vecVelocity", frame.fVelocity);
+					frame.buttons = buttons;
+					frame.iFlags = GetEntityFlags(client);
+					frame.movetype = GetEntityMoveType(client);
+
+					frame.bDucked = GetEntProp(client, Prop_Data, "m_bDucked") != 0;
+					frame.bDucking = GetEntProp(client, Prop_Data, "m_bDucking") != 0;
+					if(g_Game == Engine_CSGO)
+					{
+						frame.fDuckTime = GetEntPropFloat(client, Prop_Data, "m_flDuckAmount");
+						frame.fDuckSpeed = GetEntPropFloat(client, Prop_Data, "m_flDuckSpeed");
+					}
+					else
+					{
+						frame.fDuckTime = GetEntPropFloat(client, Prop_Data, "m_flDucktime");
+					}
+
+					gA_Frames[client].SetArray(iFrameNumber-1, frame);
+
+					gI_IndexCounter[client] = iFrameNumber - 1 - gI_PreFrameCount[client];
+					gF_IndexCounter[client] = iFrameNumber - 1.0 - view_as<float>(gI_PreFrameCount[client]);
+				}
+
+				// Fix boosters
+				// Credit to bTimes 2.0 by blacky ;)
+				if((GetEntityFlags(client) & FL_BASEVELOCITY) > 0)
+				{
+					float vBaseVel[3];
+					GetEntPropVector(client, Prop_Data, "m_vecBaseVelocity", vBaseVel);
+					
+					if(vBaseVel[2] > 0)
+					{
+						vBaseVel[2] *= 1.0 / GetEntPropFloat(client, Prop_Data, "m_flLaggedMovementValue");
+					}
+					
+					SetEntPropVector(client, Prop_Data, "m_vecBaseVelocity", vBaseVel);
+				}
+			}
+			else if(gI_Status[client] == PAUSED)
+			{
+				if(sv_client_predict != null)
+				{
+					sv_client_predict.ReplicateToClient(client, "0"); //Fix bug
+				}
+
+				if((gI_LastButtons[client] & IN_JUMP) == 0 && (buttons & IN_JUMP))
+				{
+					gI_Status[client] = RUN;
+					ResumePlayer(client);
+				}
+				else
+				{
+					vel[0] = 0.0;
+					vel[1] = 0.0;
+					vel[2] = 0.0;
+					int iFrameSize = gA_Frames[client].Length - gI_PreFrameCount[client];
+					int iFrameNumber = gI_IndexCounter[client] + gI_PreFrameCount[client];
+					if(iFrameSize > 1 && iFrameNumber > 1)
+					{
+						framedata_t frame;
+						gA_Frames[client].GetArray(iFrameNumber, frame);
+
+						float fAngles[3];
+						fAngles[0] = frame.fEyeAngles[0];
+						fAngles[1] = frame.fEyeAngles[1];
+						fAngles[2] = 0.0;
+
+						TeleportEntity(client, frame.fPosition, fAngles, view_as<float>({0.0, 0.0, 0.0}));
```

Comment:
swapping out fAngles with NULL_VECTOR here and also in ResumePlayer() would be nice. allows the player to look around while paused and easier telehopping